app/models/amr_uploaded_reading.rb

```diff
@@ -0,0 +1,75 @@
+# == Schema Information
+#
+# Table name: amr_uploaded_readings
+#
+#  amr_data_feed_config_id :bigint(8)        not null
+#  created_at              :datetime         not null
+#  file_name               :text             default("f"), not null
+#  id                      :bigint(8)        not null, primary key
+#  imported                :boolean          default(FALSE), not null
+#  reading_data            :json             not null
+#  updated_at              :datetime         not null
+#
+# Indexes
+#
+#  index_amr_uploaded_readings_on_amr_data_feed_config_id  (amr_data_feed_config_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (amr_data_feed_config_id => amr_data_feed_configs.id) ON DELETE => cascade
+#
+
+class AmrUploadedReading < ApplicationRecord
+  belongs_to :amr_data_feed_config
+
+  validates_presence_of :file_name
+
+  validate :validate_reading_data, on: :validate_reading_data
```

Comment:
Only save it if actually valid? You can still show the errors etc...

Reply:
Actually let's have a ReadingData activemodel object which we can then use anywhere....